config/initializers/012-web_hook_events.rb

```diff
@@ -74,6 +74,16 @@
   end
 end
 
+%i(
+  user_badge_granted
+).each do |event|
+  # user_badge_revoked
+  DiscourseEvent.on(event) do |badge, user_id|
+    ub = UserBadge.find_by(badge: badge, user_id: user_id)
```

Comment:
I'm fairly sure this is something like a 4*N query count on badge grant jobs, but I'm not sure how to do any better. Moving the UserBadge.find_by into the DEvent.trigger doesn't exactly help, at all.